### PR TITLE
✨ Add `InputState` and `OutputState` to `TxResult`

### LINKED ISSUE #3446: Add `InputState` and `OutputState` to `TxResult`
While overhauling `TxExecution` and `TxResult` in #3429, as a temporary measure, `UpdatedStates` and `UpdatedFungibleAssets` where removed. To provide an easier (and richer) way to access states resulting from an execution of a `Transaction`, `InputState` and `OutputState` of type `HashDigest<SHA256>?` should be provided for `TxResult`.

### PR DESCRIPTION
**Pull request by riemannulus**
Resolve #3446.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ To be released.
 ### Backward-incompatible storage format changes
 
 ### Added APIs
+ -  (Libplanet.Explorer) Added `TxResult.InputState` of type 
+    `HashDigest<SHA256>?`.  [[#3446, #3447]]
+ -  (Libplanet.Explorer) Added `TxResult.OutputState` of type 
+    `HashDigest<SHA256>?`.  [[#3446, #3447]]
```
Comment (greymistcube) [inline CHANGES.md:21]: Formatting. 😗

File: CHANGES.md [hunk 2]
```diff
@@ -25,6 +29,9 @@ To be released.
 ### CLI tools
 
 
+[#3446]: https://github.com/planetarium/libplanet/issues/3446
+[#3447]: https://github.com/planetarium/libplanet/pull/3447
+
```
Comment (greymistcube) [inline CHANGES.md:34]: Formatting. 😗

Comment (riemannulus) [reply CHANGES.md]: https://github.com/planetarium/libplanet/pull/3447/commits/b811d2d8e571c5bf12bf5820fa468fee24b10571 amended

File: CHANGES.md (hunk 1 above)
Comment (riemannulus) [reply CHANGES.md]: https://github.com/planetarium/libplanet/pull/3447/commits/b811d2d8e571c5bf12bf5820fa468fee24b10571 amended

**Review by greymistcube — APPROVED** [review]

**Pull request merged**